### PR TITLE
Html css popup

### PR DESCRIPTION
**Pull request by gorodiskiy**
_Replace task name and links as appropriate and remove this line._

### HTML-CSS-Popup
[Demo](https://gorodiskiy.github.io/kottans-frontend/tasks_html-css-popup) |
[Code base](https://github.com/gorodiskiy/kottans-frontend/tree/master/tasks_html-css-popup)

The code is submitted in its own sub-directory and in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/gorodiskiy/HTML-CSS-Popup/css/main.css [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    padding: 10px 20px;
+    background-color: #1E7B0B;
+    color: #fff;
+}
+
+input[type="checkbox"] {
+    display: none;
+}
+
+/* input[type="search"]:focus {
```
Comment (CuteShaun) [inline submissions/gorodiskiy/HTML-CSS-Popup/css/main.css]: Delete dead code, please.

File: submissions/gorodiskiy/HTML-CSS-Popup/css/main.css [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+.list__item a {
+    display: block;
+}
+
+.list__item a:hover {
+    outline: 1px solid #1E7B0B;
+}
+
+figure {
```
Comment (CuteShaun) [inline submissions/gorodiskiy/HTML-CSS-Popup/css/main.css]: Write all your resets at the top of your css.

File: submissions/gorodiskiy/HTML-CSS-Popup/css/main.css [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    height: 90vh;
+    min-height: 300px;
+    align-items: center;
+    justify-content: center;
+}
+
+.logo {
+    max-width: 150px;
+    margin-bottom: 16px;
+}
```
Comment (CuteShaun) [inline submissions/gorodiskiy/HTML-CSS-Popup/css/main.css]: Add new line at end of files, it's common practice, if you don't use it, your files can be not correct concatenate. Check this in all files, please.

File: submissions/gorodiskiy/HTML-CSS-Popup/css/main.css [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+    color: #000;
+    -webkit-box-shadow: 0 2px 10px rgba(0, 0, 0, .2);
+    box-shadow: 0 2px 10px rgba(0, 0, 0, .2);
+    display: none;
+    position: absolute;
+    right: 10px;
+    margin-top: 20px;
+}
+
+.list {
```
Comment (CuteShaun) [inline submissions/gorodiskiy/HTML-CSS-Popup/css/main.css]: What is list, list of what? Please use more specific names for your selectors.

File: submissions/gorodiskiy/HTML-CSS-Popup/css/main.css [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+    display: block;
+    text-align: center;
+    color: #1E7B0B;
+}
+
+.list--two+a {
+    display: none;
+}
+
+.list__item {
```
Comment (CuteShaun) [inline submissions/gorodiskiy/HTML-CSS-Popup/css/main.css]: .popup-list_item - more specific name, I recommend u use something like this.

File: submissions/gorodiskiy/HTML-CSS-Popup/index.html [hunk 6]
```diff
@@ -0,0 +1,141 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+    <link rel="stylesheet" href="css/main.css" />
+    <title>Document</title>
```
Comment (CuteShaun) [inline submissions/gorodiskiy/HTML-CSS-Popup/index.html]: Please, use name for your document

File: submissions/gorodiskiy/HTML-CSS-Popup/index.html [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+                ><img src="img/icon-apps.png" alt="icon-apps"
+              /></label>
+              <input type="checkbox" name="open-box" id="open-box" />
+              <div class="box">
+                <input type="checkbox" name="more" id="more" />
+                <ul class="list">
+                  <li class="list__item">
+                    <a href="#">
+                      <figure>
+                        <p><img src="img/apps-blogger.png" alt="Blog" /></p>
```
Comment (CuteShaun) [inline submissions/gorodiskiy/HTML-CSS-Popup/index.html]: Why you use p tag for image? I guess div more proper for this case))

File: submissions/gorodiskiy/HTML-CSS-Popup/index.html [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+                        <figcaption>Blog</figcaption>
+                      </figure>
+                    </a>
+                  </li>
+                  <li class="list__item">
+                    <a href="#">
+                      <figure>
+                        <p><img src="img/apps-booble+.png" alt="Booble+" /></p>
```
Comment (CuteShaun) [inline submissions/gorodiskiy/HTML-CSS-Popup/index.html]: See comment before.

File: submissions/gorodiskiy/HTML-CSS-Popup/index.html [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+                      <figure>
+                        <p><img src="img/apps-map.png" alt="Map" /></p>
+                        <figcaption>Map</figcaption>
+                      </figure>
+                    </a>
+                  </li>
+                  <li class="list__item">
+                    <a href="#">
+                      <figure>
+                        <p><img src="img/apps-market.png" alt="Market" /></p>
```
Comment (CuteShaun) [inline submissions/gorodiskiy/HTML-CSS-Popup/index.html]: See comment before.

File: submissions/gorodiskiy/HTML-CSS-Popup/css/main.css [hunk 10]
```diff
@@ -0,0 +1,187 @@
+* {
+    box-sizing: border-box;
+    -webkit-appearance: none;
```
Comment (A-Ostrovnyy) [inline submissions/gorodiskiy/HTML-CSS-Popup/css/main.css]: This is not a good rule for style reset.
This rule turned off the default behavior of all elements, better applied it pointwise for the required elements.

File: submissions/gorodiskiy/HTML-CSS-Popup/css/main.css [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+    display: flex;
+    justify-content: flex-end;
+    padding: 0 20px;
+}
+
+header {
+    padding: 5px 0;
+}
+
+#open-box:checked~.box {
```
Comment (A-Ostrovnyy) [inline submissions/gorodiskiy/HTML-CSS-Popup/css/main.css]: Avoid styling by `id`. 
Separate responsibilities: HTML for markup, CSS for visual representing, ID - for javascript

File: submissions/gorodiskiy/HTML-CSS-Popup/css/main.css [hunk 12]
```diff
@@ -184,4 +181,6 @@ figure img {
 .logo {
     max-width: 150px;
     margin-bottom: 16px;
-}
\ No newline at end of file
+}
+
+/*  */
```
Comment (A-Ostrovnyy) [inline submissions/gorodiskiy/HTML-CSS-Popup/css/main.css]: Remove all content on line 186 and leave all rest.
This red sign([Image](http://prntscr.com/prmvhj)) means that you don't have an empty line at the end of file.

**Comment by A-Ostrovnyy** [comment]
Add styles on ":hover" state for a button more.

**Comment by gorodiskiy** [comment]
> Add styles on ":hover" state for a button more.

Done )

**Comment by A-Ostrovnyy** [comment]
Great!
But still can't open a popup by keyboard(

**Comment by gorodiskiy** [comment]
Done Keyboard Accessibility for Popups

File: submissions/gorodiskiy/HTML-CSS-Popup/index.html [hunk 13]
```diff
@@ -138,4 +138,4 @@
   </div>
 </body>
 
-</html>
+</html>
```
Comment (A-Ostrovnyy) [inline submissions/gorodiskiy/HTML-CSS-Popup/index.html]: Don't forget about an empty line at the end of each file. This option can be automated by your code editor check settings.

**Comment by gorodiskiy** [comment]
Done )
the second popup is also responsive from the keyboard

File: submissions/gorodiskiy/HTML-CSS-Popup/css/main.css [hunk 14]
```diff
@@ -91,7 +91,14 @@ input[name="open-box"]{
 input[name="open-box"]:checked~.box {
     display: block;
 }
+input[name="more"]{
+  position: absolute;
+  height:40px;
+  width:100%;
+  display: block;
+  margin: -40px 0 0 0;
```
Comment (A-Ostrovnyy) [inline submissions/gorodiskiy/HTML-CSS-Popup/css/main.css]: Watch the [video](https://www.youtube.com/watch?v=E6kLaaQFctU&t=413s) to find out how to hide checkbox in right way.
Bring back styles on `:focus` state.

Comment (gorodiskiy) [reply submissions/gorodiskiy/HTML-CSS-Popup/css/main.css]: Yes, but how will I open the second popup then? if my input will lie inside the label, I will not be able to access ul.
Or embed ul in label too

label> input + ul

Comment (A-Ostrovnyy) [reply submissions/gorodiskiy/HTML-CSS-Popup/css/main.css]: Use this markup:
```HTML
<input id="someId">
<label for="someId"></label>
<ul>
  <li></li>
</ul>
```
```CSS
input:focused + label {
    rule: value;
}

input:checked ~ ul {
    display: block;
}
```
Hope it helps you. Read more about [sibling combinator](https://developer.mozilla.org/en-US/docs/Web/CSS/General_sibling_combinator)

Comment (gorodiskiy) [reply submissions/gorodiskiy/HTML-CSS-Popup/css/main.css]: Yes, just an example of Makeev, there is an input inside the label. Maybe I understood you too literally

Comment (A-Ostrovnyy) [reply submissions/gorodiskiy/HTML-CSS-Popup/css/main.css]: It doesn't matter where input in, an inside label or outside. The correct way to hide checkbox/radio button is using rules from the class `.visually-hidden` or `appearance: none;` to input.)

**Comment by A-Ostrovnyy** [comment]
Well done!

**Pull request merged**